packages/reqresp/src/ReqResp.ts

```diff
@@ -48,29 +51,52 @@ export class ReqResp {
     this.logger = modules.logger;
     this.metrics = modules.metricsRegister ? getMetrics(modules.metricsRegister) : null;
     this.protocolPrefix = opts.protocolPrefix ?? DEFAULT_PROTOCOL_PREFIX;
+    this.rateLimiter = new ReqRespRateLimiter(
+      {
+        metrics: this.metrics,
+        reportPeer: modules.reportPeer,
+        logger: modules.logger,
+      },
+      {
+        rateLimitMultiplier: opts.rateLimitMultiplier,
+      }
+    );
   }
 
   registerProtocol<Req, Resp>(protocol: ProtocolDefinition<Req, Resp>): void {
     const {method, version, encoding} = protocol;
     const protocolID = this.formatProtocolID(method, version, encoding);
     this.supportedProtocols.set(protocolID, protocol as ProtocolDefinition);
+    this.rateLimiter.initRateLimits(protocol);
   }
 
   async start(): Promise<void> {
     this.controller = new AbortController();
+    this.rateLimiter.start();
     // We set infinity to prevent MaxListenersExceededWarning which get logged when listeners > 10
     // Since it is perfectly fine to have listeners > 10
     setMaxListeners(Infinity, this.controller.signal);
 
     for (const [protocolID, protocol] of this.supportedProtocols) {
-      await this.libp2p.handle(protocolID, this.getRequestHandler(protocol));
+      try {
+        await this.libp2p.handle(protocolID, this.getRequestHandler(protocol));
```

Comment:
What the rationale of this code change? What does this have to do with rate limit?